app/support/journals/closer.rb

```diff
@@ -47,17 +47,25 @@ def mark_as_succeeded_with_errors
 
   def mark_as_succeeded
     if journal.update_attributes(params.merge(is_successful: true))
-      journal.order_details.update_all(
+      result = journal.order_details.update_all(
```

Comment:
If we do this, and `after_success` returns `false`, wouldn't that change the behavior of this function? I am concerned that would break backwards compatibility.

My current mental model is that this function returns true if the journal was closed successfully.

If we return false in a case where the journal was closed successfully, but the `after_success` failed, then would that possibly cause unintended behavior in callers of this function?

Reply:
Good question.  The `update_all` method generates an SQL statement to update all the records according to the params passed in - it doesn't run model validations, so the chance of failure is very low (but not nil).  It returns an integer, the number of records that were updated, and in ruby integers are truthy.  

The return value of `mark_as_succeeded` ultimately gets used to decide what the user will see after attempting to close the journal - either a success message or no success message (see `FacilityJournalsController#update`, line 69). I don't think we want to change that behavior.  This makes me lean towards making the change in `FacilityJournalsController#update`.  See my comment below for more info.